src/map/status.c

```diff
@@ -11114,18 +11263,192 @@ static int status_change_end_(struct block_list *bl, enum sc_type type, int tid,
 					break;
 			}
 			break;
-		case SC_ARMOR_RESIST:
-			if (sce->val1 > 0)
-				clif->sc_end(bl, bl->id, AREA, status->get_sc_icon(SC_RESIST_PROPERTY_WATER));
+		case SC_ARMOR_RESIST: {
+			struct status_change_entry *sce_water = sc->data[SC_RESIST_PROPERTY_WATER];
+			struct status_change_entry *sce_ground = sc->data[SC_RESIST_PROPERTY_GROUND];
+			struct status_change_entry *sce_fire = sc->data[SC_RESIST_PROPERTY_FIRE];
+			struct status_change_entry *sce_wind = sc->data[SC_RESIST_PROPERTY_WIND];
 
-			if (sce->val2 > 0)
-				clif->sc_end(bl, bl->id, AREA, status->get_sc_icon(SC_RESIST_PROPERTY_GROUND));
+			// Water
+			clif->sc_end(bl, bl->id, AREA, status->get_sc_icon(SC_RESIST_PROPERTY_WATER));
+			if (sce_water != NULL) {
+				int sum_water = sce_water->val1 + ((sce_fire != NULL) ? sce_fire->val2 : 0);
+				if (sce_water->val1 > 0 && sum_water > 0) {
+					const struct TimerData *td = timer->get(sce_water->timer);
```

Comment:
``timer->get`` may return NULL. also sce_water->timer can be invalid timer. probably need add checks for timer field and return value of ``timer->get``?